stdlib/3/pathlib.pyi

```diff
@@ -95,7 +95,10 @@ class Path(PurePath):
              encoding: Optional[str] = ..., errors: Optional[str] = ...,
              newline: Optional[str] = ...) -> IO[Any]: ...
     def owner(self) -> str: ...
-    def rename(self, target: Union[str, PurePath]) -> None: ...
+    if sys.version_info < (3, 8):
+        def rename(self, target: Union[str, PurePath]) -> None: ...
+    else:
+        def rename(self, target: Union[str, PurePath]) -> _P: ...
```

Comment:
I think you need `self: _P` here; currently the typevar remains unbound. I haven't checked though to make sure that the returned type is actually always the same as the self type.

Also, it's slightly cleaner to put the newer version first, like on line 111/117 below.

Reply:
`self: _P` looks correct from what I can see in other similar functions in the same file such as `with_name` and `with_suffix`. I'll add that.

I can see this specific file is a bit inconsistent with which order the versions are listed in. The first instance of the file has the lowest version first which is what I based this on: https://github.com/python/typeshed/blob/f5a1925e765b92dd1b12ae10cf8bff21c225648f/stdlib/3/pathlib.pyi#L22-L27
Would you like me to make the ordering consistent in the rest of the file as well?

Reply:
Thanks! The rest of the file can wait for antoher day.